src/Controls/src/Core/HandlerImpl/CheckBox/CheckBox.Impl.cs

```diff
@@ -12,6 +12,8 @@ public partial class CheckBox : ICheckBox, IMapColorPropertyToPaint
 		void IMapColorPropertyToPaint.MapColorPropertyToPaint(Color color)
 		{
 			Foreground = color?.AsPaint();
```

Comment:
Doesn't this automatically trigger the Foreground property changed?

Reply:
Yes, but in the wrong order. This is thrown first, and then the property is updated.

Reply:
Oh that interface looks special... I never saw it before. And if going the simpler and used in other places way works then maybe that is the best way for now?

Reply:
Just sent an alternative removing `IMapColorPropertyToPaint` here https://github.com/dotnet/maui/pull/12102